web-handlers.go

```diff
@@ -443,3 +336,104 @@ func (web *webAPI) SetAuth(r *http.Request, args *SetAuthArgs, reply *SetAuthRep
 	reply.UIVersion = miniobrowser.UIVersion
 	return nil
 }
+
+// GetAuthReply - Reply current credentials.
+type GetAuthReply struct {
+	AccessKey string `json:"accessKey"`
+	SecretKey string `json:"secretKey"`
+	UIVersion string `json:"uiVersion"`
+}
+
+// GetAuth - return accessKey and secretKey credentials.
+func (web *webAPI) GetAuth(r *http.Request, args *GenericArgs, reply *GetAuthReply) error {
+	if !isJWTReqAuthenticated(r) {
+		return &json2.Error{Message: "Unauthorized request"}
+	}
+	creds := serverConfig.GetCredential()
+	reply.AccessKey = creds.AccessKeyID
+	reply.SecretKey = creds.SecretAccessKey
+	reply.UIVersion = miniobrowser.UIVersion
+	return nil
+}
+
+// Upload - file upload handler.
+func (web *webAPI) Upload(w http.ResponseWriter, r *http.Request) {
+	if !isJWTReqAuthenticated(r) {
+		writeErrorResponse(w, r, ErrSignatureDoesNotMatch, r.URL.Path)
+		return
+	}
+	vars := mux.Vars(r)
+	bucket := vars["bucket"]
+	object := vars["object"]
+	if _, err := web.Filesystem.CreateObject(bucket, object, -1, r.Body, nil); err != nil {
+		writeWebErrorResponse(w, err)
+	}
+}
+
+// Download - file download handler.
+func (web *webAPI) Download(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	bucket := vars["bucket"]
+	object := vars["object"]
+	token := r.URL.Query().Get("token")
+
+	jwt := initJWT()
+	jwttoken, e := jwtgo.Parse(token, func(token *jwtgo.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwtgo.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		return []byte(jwt.SecretAccessKey), nil
+	})
+	if e != nil || !jwttoken.Valid {
+		writeErrorResponse(w, r, ErrSignatureDoesNotMatch, r.URL.Path)
```

Comment:
Let's not write back XML here, you can write it like this.

```
writeWebErrorResponse(w, errInvalidToken) 
```

errInvalidToken define it in typed-errors.go and then you can make it like. 

```
if e == errInvalidToken {
    w.WriteHeader(http.StatusBadRequest)
    w.Write([]byte(errInvalidToken.Error()))
```